File: ServerSpace/src/main/java/DataLayer/ServicioDB.java
package DataLayer;

import EntityLayer.ServicioEntity;
import Enumerados.ProcessActionEnum;
import Framework.BaseDB;
import Framework.Inj;
import Framework.Utilidades;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;

public class ServicioDB extends BaseDB {
    public ArrayList<ServicioEntity> ObtenerMain() {

        ArrayList<ServicioEntity> DatoMemoria = new ArrayList<>();
        ServicioEntity en;
        try {
            Inj.Sp("sp_ServicioObtenerMain");
            ResultSet rs = Inj.RunSelect();
            fillSchemeTable(rs);
            while (rs.next()) {
                en = new ServicioEntity();
                if (fillFrom(rs, en)) {
                    DatoMemoria.add(en);
                }
            }

        } catch (SQLException e) {
            System.out.println("ERROR " + e);
            throw new UnsupportedOperationException("Datalater :" + e);
        }
        return DatoMemoria;
    }

    public ServicioEntity Save(ServicioEntity entity) {
        Boolean State = null;
        try {
            Inj.IniciarTranssaccion(false);
            String Store = "sp_ServicioRegistrar";
            if (entity.getAction() == ProcessActionEnum.Update.getValor()) {
                Store = "sp_ServicioActualizar";
            }
            Inj.Sp(Store);
            Inj.Pmt_Integer("@ServicioId", entity.getServicioId(), true);
            Inj.Pmt_String("@Nombre", entity.getNombre(), false);
            Inj.Pmt_String("@Descripcion", entity.getDescripcion(), false);
            Inj.Pmt_Integer("@TipoServicioId", entity.getTipoServicioId(), false);
            Inj.Pmt_String("@FechaRegistro", Utilidades.getFechaRegistro(), false);
            Inj.Pmt_String("@CodUsuario", entity.getCodUsuario(), false);
            Inj.Pmt_Boolean("@EstadoRegistro", entity.getEstadoRegistro(), false);
            if (entity.getAction() == ProcessActionEnum.Add.getValor()) {
                int Id = Inj.RunInsert();
                State = Id > 0;
                if (State) {
                    entity.setServicioId(Id);
                }
            }
            if (entity.getAction() == ProcessActionEnum.Update.getValor()) {
                Inj.RunUpdate();
            }

        } catch (Exception ex) {
            throw new UnsupportedOperationException("Datalater : " + ex);
        }
        return entity;
    }

    public ArrayList<ServicioEntity> ObtenerItem(int ServicioId) {

        ArrayList<ServicioEntity> DatoMemoria = new ArrayList<>();
        ServicioEntity en;
        try {
            Inj.Sp("sp_ServicioObtenerItem");
            Inj.Pmt_Integer("@ServicioId", ServicioId, false);
            ResultSet rs = Inj.RunSelect();
            fillSchemeTable(rs);
            while (rs.next()) {
                en = new ServicioEntity();
                if (fillFrom(rs, en)) {
                    DatoMemoria.add(en);
                }
            }

        } catch (SQLException e) {
            System.out.println("ERROR " + e);
            throw new UnsupportedOperationException("Datalater :" + e);
        }
        return DatoMemoria;
    }
}

File: Space/Space.DataLayer/ServicioDB.cs
using Framework;
using Space.EntityLayer;
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Space.DataLayer
{
    public class ServicioDB : BaseDB
    {
        public List<ServicioEntity> ObtenerMain()
        {
            try
            {
                List<ServicioEntity> ListaData = new List<ServicioEntity>();
                Conexion.OpenConnection();
                Conexion.Store("sp_ServicioObtenerMain");
                SqlDataReader reader = Conexion.ReaderData();
                FillSchemeTable(reader);
                while (reader.Read())
                {
                    ServicioEntity entity = new ServicioEntity();
                    if (FillFrom<ServicioEntity>(reader, entity)) ListaData.Add(entity);
                }
                reader.Close();
                Conexion.CloseConnection();
                return ListaData;

            }
            catch (Exception ex)
            {
                Conexion.Restore();
                throw ex;
            }
        }

        public Int32 Registrar(ServicioEntity Ent)
        {
            try
            {

                String NameStore = "sp_ServicioRegistrar";

                Conexion.Store(NameStore);
                Conexion.AddParemeter("ServicioId", SqlDbType.Int, 10, ParameterDirection.InputOutput, Ent.ServicioId);
                Conexion.AddParemeter("Nombre", SqlDbType.VarChar, 150, ParameterDirection.Input, Ent.Nombre);
                Conexion.AddParemeter("Descripcion", SqlDbType.VarChar, 250, ParameterDirection.Input, Ent.Descripcion);
                Conexion.AddParemeter("TipoServicioId", SqlDbType.Int, 10, ParameterDirection.Input, Ent.TipoServicioId);
                Conexion.AddParemeter("FechaRegistro", SqlDbType.DateTime, 15, ParameterDirection.Input, Ent.FechaRegistro);
                Conexion.AddParemeter("CodUsuario", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.CodUsuario);
                Conexion.AddParemeter("EstadoRegistro", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.EstadoRegistro);

                Ent.ServicioId = Conexion.ExecuteQueryItem("ServicioId");
                return Ent.ServicioId;
            }
            catch (Exception ex)
            {
                Conexion.Restore();
                throw ex;
            }
        }


        public Int32 Actualizar(ServicioEntity Ent)
        {
            try
            {

                String NameStore = "sp_ServicioActualizar";

                Conexion.Store(NameStore);
                Conexion.AddParemeter("ServicioId", SqlDbType.Int, 10, ParameterDirection.InputOutput, Ent.ServicioId);
                Conexion.AddParemeter("Nombre", SqlDbType.VarChar, 150, ParameterDirection.Input, Ent.Nombre);
                Conexion.AddParemeter("Descripcion", SqlDbType.VarChar, 250, ParameterDirection.Input, Ent.Descripcion);
                Conexion.AddParemeter("TipoServicioId", SqlDbType.Int, 10, ParameterDirection.Input, Ent.TipoServicioId);
                Conexion.AddParemeter("FechaRegistro", SqlDbType.DateTime, 15, ParameterDirection.Input, Ent.FechaRegistro);
                Conexion.AddParemeter("CodUsuario", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.CodUsuario);
                Conexion.AddParemeter("EstadoRegistro", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.EstadoRegistro);

                Ent.ServicioId = Conexion.ExecuteQueryItem("ServicioId");
                return Ent.ServicioId;
            }
            catch (Exception ex)
            {
                Conexion.Restore();
                throw ex;
            }
        }

        public List<ServicioEntity> ObtenerItem(Int32 ServicioId)
        {
            try
            {
                List<ServicioEntity> ListaData = new List<ServicioEntity>();
                Conexion.OpenConnection();
                Conexion.Store("sp_ServicioObtenerItem");
                Conexion.AddParemeter("ServicioId", SqlDbType.Int, 10, ParameterDirection.Input, ServicioId);
                SqlDataReader reader = Conexion.ReaderData();
                FillSchemeTable(reader);
                while (reader.Read())
                {
                    ServicioEntity entity = new ServicioEntity();
                    if (FillFrom<ServicioEntity>(reader, entity)) ListaData.Add(entity);
                }
                reader.Close();
                return ListaData;

            }
            catch (Exception ex)
            {
                Conexion.Restore();
                throw ex;
            }
        }


    }
}
